src/solver/tensor/Op2dTensorLite.cpp

```diff
@@ -0,0 +1,217 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/tensor/solvers.hpp>
+
+#include <miopen/tensor/invoke_params.hpp>
+#include <miopen/tensor.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/float_equal.hpp>
+#include <miopen/datatype.hpp>
+
+namespace miopen {
+
+namespace solver {
+
+namespace tensor {
+
+bool Op2dTensorLite::IsApplicable(const ExecutionContext& context,
+                                  const miopen::tensor::ProblemDescription& problem) const
+{
+    auto aTensorDesc = problem.GetATensorDesc();
+    auto bTensorDesc = problem.GetBTensorDesc();
+    auto cTensorDesc = problem.GetCTensorDesc();
+
+    auto alens = aTensorDesc.GetLengths();
+    auto blens = bTensorDesc.GetLengths();
+    auto clens = cTensorDesc.GetLengths();
+
+    auto asize = alens.size();
+
+    size_t local_threads = 256;
+    int max_num_wg       = 4096;
+
+    // for naive tensor ops
+    size_t RD_BLCK    = (clens[2] % 4 == 0) ? 4 : (clens[2] % 2 == 0) ? 2 : 1;
+    size_t total_work = std::max(clens[2] / RD_BLCK, size_t(1));
+    size_t grp_sz     = (total_work + local_threads - 1) / local_threads;
```

Comment:
It looks common for `isApplicable` and `getSolution` and probably for the other solvers.